test/functional/test_framework/util.py

```diff
@@ -208,10 +208,10 @@ def str_to_b64str(string):
 def satoshi_round(amount):
     return Decimal(amount).quantize(Decimal('0.00000001'), rounding=ROUND_DOWN)
 
-def wait_until(predicate, *, attempts=float('inf'), timeout=float('inf'), lock=None, factor=1.0):
+def wait_until(predicate, *, attempts=float('inf'), timeout=float('inf'), lock=None, timeout_factor=1.0):
```

Comment:
(for a follow-up PR) this is a really weird interface. The client is saying: 'take these two numbers, multiply them and use that as the timeout'. The client is able to do that multiplication itself! Why does it need to pass in two different values?

Reply:
Thanks for the review. The process of making a PR was educational and made me go deeper into stuffs. Couldn't have done without your review club.
   
And also thanks for the follow-up idea. Yes it does seem redundant. Surely this can be improved. I will look into this. :+1:

Reply:
On an initial look it seems this is fairly trivial. There is no test function that calls `wait_until()` with `timeout_factor`. Either uses default or specifies a specific timeout. The only place `timeout_factor` is used is at node initialization and peer connection logic, which already knows how to handle it. 
So it seems it just requires changing the `wait_until()` signature.
Let me know if I am missing something there.